dartagnan/src/main/java/com/dat3m/dartagnan/program/ScopeHierarchy.java

```diff
@@ -32,6 +32,14 @@ public static ScopeHierarchy ScopeHierarchyForPTX(int gpu, int cta) {
         return scopeHierarchy;
     }
 
+    public static ScopeHierarchy ScopeHierarchyForOpenCL(int dev, int wg) {
+        ScopeHierarchy scopeHierarchy = new ScopeHierarchy();
+        scopeHierarchy.scopeIds.put(Tag.OpenCL.ALL, 0);
+        scopeHierarchy.scopeIds.put(Tag.OpenCL.DEVICE, dev);
+        scopeHierarchy.scopeIds.put(Tag.OpenCL.WORK_GROUP, wg);
+        return scopeHierarchy;
+    }
```

Comment:
Somewhere you were defining a hierarchy with two elements (dev and wg) and now there are three. Shouldn't there be a match between that in this?

Reply:
I just noticed this is analogous to `ScopeHierarchyForVulkan`, thus it is fine